lib/web_ui/lib/src/engine/compositor/rasterizer.dart

```diff
@@ -2,38 +2,25 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-// @dart = 2.6
+
 part of engine;
 
 /// A class that can rasterize [LayerTree]s into a given [Surface].
 class Rasterizer {
   final Surface surface;
   final CompositorContext context = CompositorContext();
-  final HtmlViewEmbedder viewEmbedder = HtmlViewEmbedder();
   final List<ui.VoidCallback> _postFrameCallbacks = <ui.VoidCallback>[];
 
-  Rasterizer(this.surface) {
-    surface.viewEmbedder = viewEmbedder;
-  }
+  Rasterizer(this.surface);
 
   /// Creates a new frame from this rasterizer's surface, draws the given
   /// [LayerTree] into it, and then submits the frame.
   void draw(LayerTree layerTree) {
     try {
-      if (layerTree == null) {
-        return;
-      }
-
-      final ui.Size physicalSize = ui.window.physicalSize;
-      final ui.Size frameSize = ui.Size(
-        physicalSize.width.truncate().toDouble(),
-        physicalSize.height.truncate().toDouble(),
-      );
-
-      if (frameSize.isEmpty) {
+      if (layerTree.frameSize.isEmpty) {
```

Comment:
this never sets the `frameSize` on `layerTree`

Reply:
In order to de-null `frameSize` I changed the initialization sequence a little. `frameSize` is now final non-null field on LayerTree, so it's guaranteed to be available.

Reply:
I also removed `frameSize` from `Rasterizer` as it is available in the `LayerTree`, which `Rasterizer` has access to already.